### ISSUE TITLE
Status of project

### ISSUE THREAD

**Issue by mikeaustin** [pedronauck/react-adopt#36]
I don't like React hooks. I understand them, and I understand the benefit in some situations, but have you written an async componentDidMount() type hook? It's a mess with nested closures and ending with empty brackets:

useEffect((() => async () => ...)(), [])

The intent of that is not clear at all, compared to the very descriptive "async componentWillMount()".

One of the "problems" hooks solve is actually not a problem - render props. When composed, I like this pattern better than hooks.

To sum up my rant, I was curious about the state of this project. It seems like the perfect solution.

 I also use Unstated, a simple library to share state between components. It has a similar way of composing containers.
